src/r3/atomic-swap/corda/evm-interop-workflows/src/main/kotlin/com/r3/corda/evminterop/workflows/swap/RequestBlockHeaderProofsInitiator.kt

```diff
@@ -0,0 +1,66 @@
+package com.r3.corda.evminterop.workflows.swap
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.evminterop.dto.TransactionReceipt
+import com.r3.corda.evminterop.workflows.eth2eth.GetBlockFlow
+import net.corda.core.crypto.DigitalSignature
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.utilities.unwrap
+import org.web3j.utils.Numeric
+
+typealias RequestBlockHeaderProofsInitiator = RequestBlockHeaderProofs.Initiator
+
+object RequestBlockHeaderProofs {
```

Comment:
It looks like this class at present is a stop-gap which polls each of the validators in turn and gathers a signature from each, while the validators themselves simply sign without actually checking anything? If so, could we have a comment indicating that it doesn't represent the complete solution?

If we want to obtain signatures from n-of-m validators in parallel, it will not be possible to do so with a single flow; do we have an answer yet for how an asynchronous pattern with each validator initiating a new flow with a callback will fit into the overall implementation?

Reply:
Yes, the next PR has the removal of the stop gap as one of its goals.